Sources/NIOTransportServices/NIOTSConnectionChannel.swift

```diff
@@ -240,89 +255,36 @@ internal final class NIOTSConnectionChannel {
-        return self._pipeline
-    }
-
-    /// The local address for this channel.
-    public var localAddress: SocketAddress? {
-        return self._addressCacheLock.withLock {
-            return self._addressCache.local
-        }
-    }
-
-    /// The remote address for this channel.
-    public var remoteAddress: SocketAddress? {
-        return self._addressCacheLock.withLock {
-            return self._addressCache.remote
-        }
+    func getChannelSpecificOption0<Option>(option: Option) throws -> Option.Value where Option : ChannelOption {
+        fatalError("option \(type(of: option)).\(option) not supported")
     }
 
-    /// Whether this channel is currently writable.
-    public var isWritable: Bool {
-        return self.backpressureManager.writable.load(ordering: .relaxed)
-    }
-
-    public var _channelCore: ChannelCore {
-        return self
-    }
-
-    public func setOption<Option: ChannelOption>(_ option: Option, value: Option.Value) -> EventLoopFuture<Void> {
-        if self.eventLoop.inEventLoop {
-            return self.eventLoop.makeCompletedFuture(Result { try setOption0(option: option, value: value) })
-        } else {
-            return self.eventLoop.submit { try self.setOption0(option: option, value: value) }
-        }
-    }
-
-    private func setOption0<Option: ChannelOption>(option: Option, value: Option.Value) throws {
+    func setChannelSpecificOption0<Option: ChannelOption>(option: Option, value: Option.Value) throws {
         self.eventLoop.preconditionInEventLoop()
 
         guard !self.closed else {
             throw ChannelError.ioOnClosedChannel
         }
 
         switch option {
-        case _ as ChannelOptions.Types.AutoReadOption:
-            self.options.autoRead = value as! Bool
-            self.readIfNeeded0()
         case _ as ChannelOptions.Types.AllowRemoteHalfClosureOption:
             self.options.supportRemoteHalfClosure = value as! Bool
-        case _ as ChannelOptions.Types.SocketOption:
-            let optionValue = option as! ChannelOptions.Types.SocketOption
-
-            // SO_REUSEADDR and SO_REUSEPORT are handled here.
-            switch (optionValue.level, optionValue.name) {
-            case (SOL_SOCKET, SO_REUSEADDR):
-                self.reuseAddress = (value as! SocketOptionValue) != Int32(0)
-            case (SOL_SOCKET, SO_REUSEPORT):
-                self.reusePort = (value as! SocketOptionValue) != Int32(0)
-            default:
-                try self.tcpOptions.applyChannelOption(option: optionValue, value: value as! SocketOptionValue)
-            }
-        case _ as ChannelOptions.Types.WriteBufferWaterMarkOption:
-            if self.backpressureManager.writabilityChanges(whenUpdatingWaterMarks: value as! ChannelOptions.Types.WriteBufferWaterMark) {
-                self.pipeline.fireChannelWritabilityChanged()
-            }
         case _ as NIOTSChannelOptions.Types.NIOTSWaitForActivityOption:
```

Comment:
This appears to be left-over, we'll never get here as it's handled in `StateManagedNWConnectionChannel`.